public/app/core/components/RolePickerDrawer/RolePickerSelect.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { MultiSelect } from '@grafana/ui';
+
+import { RolePickerDrawer } from './RolePickerDrawer';
+
+export interface Props {}
+
+export const RolePickerSelect = ({}: Props): JSX.Element => {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const { control } = useFormContext();
+
+  const toggleDrawer = () => setIsDrawerOpen(!isDrawerOpen);
+
+  return (
+    <>
+      <Controller
+        name="role-collection"
+        control={control}
+        render={({ field: { ref, value, ...field } }) => (
+          <MultiSelect
+            {...field}
+            onOpenMenu={toggleDrawer}
+            onChange={() => {
+              // TODO cannnot remove basic roles
+              // TODO open drawer instead
+            }}
+            value={value?.map((role: string) => ({ label: role, value: role }))}
```

Comment:
```suggestion
            value={value?.map(toOption)}
```
There's a `toOption` util from `@grafana/data` that you can use here and in the RolePickerDrawer.

Reply:
Thank you! Nice to know!